packages/perseus-score/src/widgets/widget-registry.ts

```diff
@@ -0,0 +1,108 @@
+import scoreNoop from "../util/score-noop";
+
+import scoreCategorizer from "./categorizer/score-categorizer";
+import validateCategorizer from "./categorizer/validate-categorizer";
+import scoreCSProgram from "./cs-program/score-cs-program";
+import scoreDropdown from "./dropdown/score-dropdown";
+import validateDropdown from "./dropdown/validate-dropdown";
+import scoreExpression from "./expression/score-expression";
+import validateExpression from "./expression/validate-expression";
+import scoreGrapher from "./grapher/score-grapher";
+import scoreGroup from "./group/score-group";
+import validateGroup from "./group/validate-group";
+import scoreIframe from "./iframe/score-iframe";
+import scoreInputNumber from "./input-number/score-input-number";
+import scoreInteractiveGraph from "./interactive-graph/score-interactive-graph";
+import scoreLabelImage from "./label-image/score-label-image";
+import validateLabelImage from "./label-image/validate-label-image";
+import scoreMatcher from "./matcher/score-matcher";
+import scoreMatrix from "./matrix/score-matrix";
+import validateMatrix from "./matrix/validate-matrix";
+import scoreMockWidget from "./mock-widget/score-mock-widget";
+import scoreNumberLine from "./number-line/score-number-line";
+import validateNumberLine from "./number-line/validate-number-line";
+import scoreNumericInput from "./numeric-input/score-numeric-input";
+import scoreOrderer from "./orderer/score-orderer";
+import validateOrderer from "./orderer/validate-orderer";
+import scorePlotter from "./plotter/score-plotter";
+import validatePlotter from "./plotter/validate-plotter";
+import scoreRadio from "./radio/score-radio";
+import validateRadio from "./radio/validate-radio";
+import scoreSorter from "./sorter/score-sorter";
+import validateSorter from "./sorter/validate-sorter";
+import scoreTable from "./table/score-table";
+import validateTable from "./table/validate-table";
+
+import type {
+    WidgetScorerFunction,
+    WidgetValidatorFunction,
+} from "../validation.types";
+
+const widgets = {};
+
+function registerWidget(
+    type: string,
+    scorer: WidgetScorerFunction,
+    validator?: WidgetValidatorFunction,
+) {
+    widgets[type] = {
+        scorer,
+        validator,
+    };
+}
+
+export const getWidgetValidator = (
+    name: string,
+): WidgetValidatorFunction | null => {
+    return widgets[name]?.validator ?? null;
+};
+
+export const getWidgetScorer = (name: string): WidgetScorerFunction | null => {
+    return widgets[name]?.scorer ?? null;
+};
+
+registerWidget(
```

Comment:
Jeremy and I are talking about merging PerseusScore and PerseusCore so there's only one registry for the logic (along with a registry for the editors and a registry for the UI). In the meantime...